api/models/endpoint.go

```diff
@@ -0,0 +1,116 @@
+package models
+
+import (
+    "github.com/frain-dev/convoy/datastore"
+    m "github.com/frain-dev/convoy/internal/pkg/middleware"
+    "github.com/frain-dev/convoy/util"
+    "net/http"
+    "strings"
+)
+
+type CreateEndpoint struct {
+    URL                string `json:"url" valid:"required~please provide a url for your endpoint"`
+    Secret             string `json:"secret"`
+    OwnerID            string `json:"owner_id"`
+    Description        string `json:"description"`
+    AdvancedSignatures bool   `json:"advanced_signatures"`
+    Name               string `json:"name" valid:"required~please provide your endpointName"`
+    SupportEmail       string `json:"support_email" valid:"email~please provide a valid email"`
+    IsDisabled         bool   `json:"is_disabled"`
+    SlackWebhookURL    string `json:"slack_webhook_url"`
+
+    HttpTimeout       string                  `json:"http_timeout"`
+    RateLimit         int                     `json:"rate_limit"`
+    RateLimitDuration string                  `json:"rate_limit_duration"`
+    Authentication    *EndpointAuthentication `json:"authentication"`
+    AppID             string                  // Deprecated but necessary for backward compatibility
```

Comment:
Ideally the comment should be directly above the field name so Go docs can pick it up and so IDE can ~~strike it out~~